### PR TITLE
IGNITE-11579

### PR DESCRIPTION
**Pull request by EdShangGG**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java [hunk 1]
```diff
@@ -434,6 +435,48 @@ public void testBaselineCollectCrd() throws Exception {
...
+            new CacheConfiguration<>("garbage1").setGroupName("groupGarbage"),
+            new CacheConfiguration<>("garbage2").setGroupName("groupGarbage")));
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "find_garbage", "--port", "11212"));
+
+        assertTrue(testOut.toString().contains("garbage not found"));
+
+        testOut.reset();
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "find_garbage", "--port", "11212"));
```
Comment (glukos) [inline modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: Do you call the same command twice on purpose?

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/IgniteCacheGroupsWithRestartsTest.java [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.apache.ignite.testframework.junits.multijvm.IgniteProcessProxy;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.apache.ignite.internal.processors.cache.persistence.GridCacheDatabaseSharedManager.IGNITE_PDS_SKIP_CHECKPOINT_ON_NODE_STOP;
+
+/**
+ *
```
Comment (glukos) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/IgniteCacheGroupsWithRestartsTest.java]: Let's document what does this test class check

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceClosure.java [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.T2;
+import org.apache.ignite.internal.util.typedef.internal.CU;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.lang.IgniteCallable;
+import org.apache.ignite.resources.IgniteInstanceResource;
+import org.apache.ignite.resources.LoggerResource;
+
+/**
+ *
```
Comment (glukos) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceClosure.java]: Please add doc

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceJobResult.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import java.io.ObjectOutput;
+import java.util.Map;
+import org.apache.ignite.internal.dto.IgniteDataTransferObject;
+import org.apache.ignite.internal.util.tostring.GridToStringInclude;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ *
```
Comment (glukos) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceJobResult.java]: Please add doc

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTask.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+import org.apache.ignite.compute.ComputeJobResult;
+import org.apache.ignite.internal.processors.task.GridInternal;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.internal.visor.VisorJob;
+import org.apache.ignite.internal.visor.VisorMultiNodeTask;
+import org.apache.ignite.internal.visor.VisorTaskArgument;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ *
```
Comment (glukos) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTask.java]: Please add doc

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskArg.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+import java.util.UUID;
+import org.apache.ignite.internal.dto.IgniteDataTransferObject;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+/**
+ *
```
Comment (glukos) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskArg.java]: Please add doc

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskResult.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import java.io.ObjectOutput;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.ignite.internal.dto.IgniteDataTransferObject;
+import org.apache.ignite.internal.util.tostring.GridToStringInclude;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+/**
+ *
```
Comment (glukos) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskResult.java]: Please add doc

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java [hunk 8]
```diff
@@ -3184,7 +3290,7 @@ public int execute(List<String> rawArgs) {
                             break;
 
                         case CACHE:
-                            cache(client, args.cacheArgs());
+                            lastOperationRes = cache(client, args.cacheArgs());
```
Comment (glukos) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java]: You have changed method signature, but behavior (lastOperationRes now shared not as side effect, but as return value) is changed for this call only. Such difference may raise questions in code reader's head. Shall we change behavior for another calls as well?

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskResult.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** Results of indexes validation from node. */
+    @GridToStringInclude
+    private Map<UUID, VisorFindAndDeleteGarbageInPersistenceJobResult> result;
+
+    /** Exceptions while indexes validation from node. */
+    @GridToStringInclude
+    private Map<UUID, Exception> exceptions;
```
Comment (glukos) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskResult.java]: Did you forget to serialize this field?

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceClosure.java [hunk 10]
```diff
@@ -0,0 +1,330 @@
...
+                if (groupContext == null) {
+                    missingCacheGroups.add(grpName);
+
+                    continue;
+                }
+
+                if (groupContext.sharedGroup())
+                    grpIds.add(groupContext.groupId());
+                else
+                    log.warning("Group[name=" + grpName + "] is not shared one, it couldn't contains garbage from destroyed caches.");
```
Comment (glukos) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceClosure.java]: "couldn't contain" is correct.

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java (hunk 8 above)
Comment (EdShangGG) [reply modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java]: Would do it in another ticket.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskResult.java (hunk 9 above)
Comment (EdShangGG) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskResult.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceClosure.java (hunk 10 above)
Comment (EdShangGG) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceClosure.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java [hunk 11]
```diff
@@ -2577,6 +2654,35 @@ private CacheArguments parseAndValidateCacheArgs() {
...
+                        continue;
+                    }
+
+                    try {
+                        cacheArgs.nodeId(UUID.fromString(nextArg));
+
+                        continue;
+                    }
+                    catch (IllegalArgumentException ignored) {
+                        //No-op.
```
[the remark this answers is not in the document]
Comment (EdShangGG) [reply modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java]: Not a issue.

File: modules/core/src/test/java/org/apache/ignite/internal/commandline/CommandHandlerParsingTest.java [hunk 12]
```diff
@@ -501,4 +505,125 @@ public void testTransactionArguments() {
         assertNull(arg.getProjection());
         assertEquals(Arrays.asList("1", "2", "3"), arg.getConsistentIds());
     }
+
+    /**
+     *
```
[the remark this answers is not in the document]
Comment (EdShangGG) [reply modules/core/src/test/java/org/apache/ignite/internal/commandline/CommandHandlerParsingTest.java]: Done.

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java (hunk 1 above)
Comment (EdShangGG) [reply modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java]: Removed.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/IgniteCacheGroupsWithRestartsTest.java (hunk 2 above)
Comment (EdShangGG) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/IgniteCacheGroupsWithRestartsTest.java]: Added.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceClosure.java (hunk 3 above)
Comment (EdShangGG) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceClosure.java]: Added.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceJobResult.java (hunk 4 above)
Comment (EdShangGG) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceJobResult.java]: Added.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTask.java (hunk 5 above)
Comment (EdShangGG) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTask.java]: Added.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskArg.java (hunk 6 above)
Comment (EdShangGG) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskArg.java]: Added.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskResult.java (hunk 7 above)
Comment (EdShangGG) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/cache/VisorFindAndDeleteGarbageInPersistenceTaskResult.java]: Added.

**Comment by dspavlov** [comment]
The ticket is resolved, closing PR. 
 
Please close PRs for your resolved tickets and other non-needed PRs. 
 
Alternatively, ask a committer 
- to use GitHub UI (committer can associate accounts here: https://gitbox.apache.org/setup/ )
- or /scripts/apply_pull_request.sh from Apache Ignite source.
- or mention in commit message 'Fixes #(a number of PR)'. 
 
These methods will automatically close PR.
 
Feel free to reopen this PR if it is actual.

**Pull request closed without merging**